modal/functions.py

```diff
@@ -976,6 +980,26 @@ async def _load(resolver: Resolver, existing_object_id: Optional[str]) -> _Funct
             else:
                 proxy_id = None
 
+            # Mount point path validation for volumes and shared volumes
+            def _validate_mount_points(
+                display_name: str, volume_likes: typing.Dict[str, typing.Union[_Volume, _SharedVolume]]
+            ) -> typing.List[typing.Tuple[str, typing.Union[_Volume, _SharedVolume]]]:
+                validated = []
+                for path, vol in volume_likes.items():
+                    path = PurePath(path).as_posix()
+                    abs_path = posixpath.abspath(path)
+
+                    if path != abs_path:
+                        raise InvalidError(f"{display_name} {abs_path} must be a canonical, absolute path.")
```

Comment:
Displaying `path` instead of `abs_path` in the error makes things a bit clearer I think, since the abspath will 1. be an absolute path, and 2. contain a local prefix that the user didn't specify in the code in case they accidentally send in a relative path

Reply:
Makes sense, can do (using `abs_path` is old code, just moved around).